### PR TITLE
forgot_password

### PR DESCRIPTION
**Pull request by roman454647**

### CODE REVIEW

File: controllers/forgotPassword.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const User = require('../db/models/User')
+const tokenCreator = require('../controllers/token.js')
+const sendEmail = require('../controllers/sendEmail')
+const forgotPasswordEmail = require('../templates/forgotPasswordTemplate')
+
+exports.sendForgotPasswordEmail = (req, res) => {
+  return User.findOne({ email: req.body.email })
```
Comment (InnaKutsik) [inline controllers/forgotPassword.js]: pls move request to query


UserQuery.findUserOne(email) - not needed one?

File: controllers/register_controller.js [hunk 2]
```diff
@@ -1,29 +1,28 @@
...
+          UserQuery.createUser(req.body, result)
+          notificationMail.sendEmail(regNotification.sendRegNotification(req.body.email))
         }).then(() => res.status(200).send('new user registered'))
-        .catch((error) => res.send(error));
-    });
-};
-
+        .catch((error) => res.send(error))
+    })
+}
```
Comment (InnaKutsik) [inline controllers/register_controller.js]: забери вкладені then
у тебе є chaining

File: controllers/resetPassword.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+const jwt = require('jsonwebtoken')
+const encrypt = require('./encrypt')
+
+exports.receiveNewPassword = (req, res) => {
+  const { userId, token } = req.params
+  //const { password, userId, token } = req.body
+  const { password } = req.body
+  
+  User.findOne({ _id: userId })
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: pls move request to query

File: controllers/resetPassword.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const User = require('../db/models/User')
+
+const jwt = require('jsonwebtoken')
+const encrypt = require('./encrypt')
+
+exports.receiveNewPassword = (req, res) => {
+  const { userId, token } = req.params
+  //const { password, userId, token } = req.body
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: ?

File: controllers/resetPassword.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  const { password } = req.body
+  
+  User.findOne({ _id: userId })
+    .then(user => {
+      const secret = user.password + '-' + user.createdAt
+      const payload = jwt.decode(token, secret)
+      if (payload.userId === user.id) {
+        encrypt.hashPassword(password)
+          .then((result) => {
+            User.findOneAndUpdate({ _id: userId }, { password: result })
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: pls move request to query

File: controllers/resetPassword.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+          }).then(() => res.status(200).send('Password changed accepted'))
+          .catch((error) => res.send(error))
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: забери вкладені then
у тебе є chaining

File: controllers/token.js [hunk 7]
```diff
@@ -0,0 +1,9 @@
+const jwt =require('jsonwebtoken')
+
+exports.usePasswordHashToMakeToken = ( passwordHash, userId, createdAt) => {
+  const secret = passwordHash + '-' + createdAt
+  const token = jwt.sign({ userId }, secret, {
+    expiresIn: 3600,
```
Comment (InnaKutsik) [inline controllers/token.js]: move to const

File: routes/resetPassword.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const express = require('express')
+const resetPassword = require('../controllers/resetPassword')
+
+const router = express.Router()
+
+
```
Comment (InnaKutsik) [inline routes/resetPassword.js]: pls add //TODO

File: controllers/forgotPassword.js (hunk 1 above)
Comment (roman454647) [reply controllers/forgotPassword.js]: done

File: controllers/register_controller.js (hunk 2 above)
Comment (roman454647) [reply controllers/register_controller.js]: done

File: controllers/resetPassword.js (hunk 3 above)
Comment (roman454647) [reply controllers/resetPassword.js]: i have findUserOne

File: controllers/resetPassword.js (hunk 4 above)
Comment (roman454647) [reply controllers/resetPassword.js]: removed

Comment (roman454647) [reply controllers/resetPassword.js]: and changed to req.body which in comment

File: controllers/resetPassword.js (hunk 3 above)
Comment (roman454647) [reply controllers/resetPassword.js]: done

File: controllers/resetPassword.js (hunk 5 above)
Comment (roman454647) [reply controllers/resetPassword.js]: done

File: controllers/token.js (hunk 7 above)
Comment (roman454647) [reply controllers/token.js]: done

File: routes/resetPassword.js (hunk 8 above)
Comment (roman454647) [reply routes/resetPassword.js]: created

File: controllers/forgotPassword.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+const UserQuery = require('../db/query/UserQuery')
+const tokenCreator = require('../controllers/token.js')
+const sendEmail = require('../controllers/sendEmail')
+const forgotPasswordEmail = require('../templates/forgotPasswordTemplate')
+
+exports.sendForgotPasswordEmail = (req, res) => {
+  console.log(req.body.email)
```
Comment (InnaKutsik) [inline controllers/forgotPassword.js]: ?

File: controllers/forgotPassword.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  return UserQuery.findUserOne(req.body.email)
+    .then((user) => {
+      if (!user) {
+        return res.status(400).json({
+          message: 'No user found with that email address.',
+        })
+      }
+      const token = tokenCreator.usePasswordHashToMakeToken(user.password, user._id, user.date)
+      const url = `http://localhost:5000/resetpassword/${user._id}/${token}`
```
Comment (InnaKutsik) [inline controllers/forgotPassword.js]: Use env variable

File: controllers/forgotPassword.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
+const UserQuery = require('../db/query/UserQuery')
+const tokenCreator = require('../controllers/token.js')
+const sendEmail = require('../controllers/sendEmail')
+const forgotPasswordEmail = require('../templates/forgotPasswordTemplate')
+
+exports.sendForgotPasswordEmail = (req, res) => {
+  console.log(req.body.email)
+  return UserQuery.findUserOne(req.body.email)
```
Comment (InnaKutsik) [inline controllers/forgotPassword.js]: Дай якусь іншу назву, бо непонятно що воно шукає

File: controllers/resetPassword.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
+const UserQuery = require('../db/query/UserQuery')
+const jwt = require('jsonwebtoken')
+const encrypt = require('./encrypt')
+
+exports.receiveNewPassword = (req, res) => {
+  const { password, userId, token } = req.body
+
+  UserQuery.findUserId(userId)
+    .then(user => {
+      console.log(user)
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: ?

File: controllers/resetPassword.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+      const secret = user.password + '-' + user.createdAt
+      const payload = jwt.decode(token, secret)
+      console.log(payload)
```
Comment (InnaKutsik) [inline controllers/resetPassword.js]: ?

File: controllers/forgotPassword.js (hunk 9 above)
Comment (roman454647) [reply controllers/forgotPassword.js]: removed

File: controllers/resetPassword.js (hunk 13 above)
Comment (roman454647) [reply controllers/resetPassword.js]: removed

File: controllers/resetPassword.js (hunk 12 above)
Comment (roman454647) [reply controllers/resetPassword.js]: removed

File: controllers/forgotPassword.js (hunk 11 above)
Comment (roman454647) [reply controllers/forgotPassword.js]: changed

File: controllers/register_controller.js [hunk 14]
```diff
@@ -1,29 +1,28 @@
...
-        .then((result) => {
-          UserQuery.createUser(req.body, result);
-            notificationMail.sendEmail(regNotification.sendRegNotification(req.body.email));
-        }).then(() => res.status(200).send('new user registered'))
-        .catch((error) => res.send(error));
-    });
-};
-
+    })
+    .then((result) => {
```
Comment (InnaKutsik) [inline controllers/register_controller.js]: what will be result - res.status(400).json({
           message: 'Email already exist',	          message: 'Email already exist',
        });	        })


or encrypt.hashPassword(password)

Comment (roman454647) [reply controllers/register_controller.js]: changed

File: controllers/forgotPassword.js (hunk 10 above)
Comment (roman454647) [reply controllers/forgotPassword.js]: використав

File: controllers/auth.controller.passport.js [hunk 15]
```diff
@@ -22,12 +22,15 @@ exports.authPassport = (req, res) => {
...
       }
 
-      const tokenResult ={
-        success: true,
-        token: jwt.sign(payload, process.env.SECRET_KEY, { expiresIn: 60 * 60 }),
-      }
-
-      return res.json({ tokenResult })
+      return jwt.sign(payload, process.env.SECRET_KEY, {
+            expiresIn: 2000
```
Comment (InnaKutsik) [inline controllers/auth.controller.passport.js]: Pls move to constant

**Pull request merged**